### ISSUE TITLE
Resolve Issue With Missing Login Button - Server Component Pod

### ISSUE THREAD

**Issue by jimmybrancaccio** [gitpod-io/gitpod#8927]
### Bug description

It appears there's an odd timing issue with the server component pod that causes the Login button on the main page to not display properly.

![image0](https://user-images.githubusercontent.com/405585/159484405-47332877-4dce-42ed-bc1e-78c0b3da32a8.jpg)

Restarting the server pod seems to resolve the issue:

```
kubectl delete pod -n <namespace> -l component=server
```

This was originally reported in Discord:

* https://discord.com/channels/816244985187008514/955637344847945788
* https://discord.com/channels/816244985187008514/955632752831660103

cc. @MrSimonEmms

### Steps to reproduce

I haven't had a chance to replicate this myself so I am not sure exactly what steps are required here or what build this happens in or started happening in. Simon may be able to fill out this section better, or advise me what I can put here for paper trail purposes.

### Expected behavior

I would expect that after I log out from my Gitpod instance that I am shown a login button or link on the page I am redirected too so I can log back in.

**Comment by MrSimonEmms** [comment]
My guess with this is that either the `dashboard` or the `server` startup in an unknown order meaning that the service that the login button requires isn't available at runtime.

This should be a (fairly) simple fix:
1. identify the service - would require looking at the `if/else` that loads the login buttons on the dashboard and what makes them display
2. put a `readinessProbe` in the deployment that requires it - if it's not ready, it restarts the pod.

We must be careful that we don't end up in a `CrashLoopBackOff` situation if we have zero providers configured - this will happen if the user doesn't configure any `authProviders` in the Installer `config.yaml`

**Comment by MrSimonEmms** [comment]
This actually isn't as simple as I'd hoped. I don't actually think it's a readiness problem and it's actually that the web socket doesn't get updated when you add a new Git integration via the UI.

As a test, I just added a self-hosted GitLab instance via the integration page in the dashboard settings, logged out and there was no option to log in via that GitLab instance. As soon as I restarted the `server`, it came up.

I think this is a bug that resides with @gitpod-io/engineering-webapp
